docs/trouble-shooting.md

```diff
@@ -0,0 +1,60 @@
+# Troubleshooting Guide
+
+Common issue users might have when using Cluster API Provider for Azure.
+
+## Debugging cluster creation
+
+While cluster build out is running, you can optionally follow the controller logs in a separate window as follows:
+
+```bash
+time kubectl get po -o wide --all-namespaces -w # Watch pod creation until azure-provider-controller-manager-0 is available
+
+kubectl logs -n capz-system azure-provider-controller-manager-0 manager -f # Follow the controller logs
+```
+
+An error such as the following in the manager could point to a mismatch between a current CAPI and an old CAPZ version:
+
+```
+E0320 23:33:33.288073       1 controller.go:258] controller-runtime/controller "msg"="Reconciler error" "error"="failed to create AzureMachine VM: failed to create nic capz-cluster-control-plane-7z8ng-nic for machine capz-cluster-control-plane-7z8ng: unable to determine NAT rule for control plane network interface: strconv.Atoi: parsing \"capz-cluster-control-plane-7z8ng\": invalid syntax"  "controller"="azuremachine" "request"={"Namespace":"default","Name":"capz-cluster-control-plane-7z8ng"}
+```
+
+### Debugging workload clusters
```

Comment:
technically the above section is also for workload clusters, it might confusing to make the distinction here

Reply:
removed the distinction and added a note on how to go about looking at both